src/models/subscriptiondata.cpp

```diff
@@ -161,6 +162,15 @@ bool SubscriptionData::fromJson(const QByteArray& json) {
       return false;
     }
 
+    QJsonArray subscriptionsList = paymentData["subscriptions"].toArray();
+    for (int i = 0; i < subscriptionsList.count(); i++) {
+      if (subscriptionsList.at(i)["product_id"] ==
```

Comment:
Does this work? Don't you need to convert to a QJsonObject first?

Reply:
`paymentData["subscriptions"]` or `paymentData`, which does get converted up around L152?  This seems to be "working on my machine".